scripts/launch.fish

```diff
@@ -14,44 +14,46 @@ test -n "$ASDF_DIR"; or set ASDF_DIR "$HOME/.asdf"
 # Check if we have the asdf binary for version >= 0.16.0
 set asdf (which asdf)
 if test -n "$asdf"
-    echo "asdf executable found at $asdf. Setting ASDF_DIR=$ASDF_DIR and adding $ASDF_DATA_DIR/shims to PATH." >&2
-    # If the binary is found, set up environment for newer asdf versions
-    set -gx ASDF_DATA_DIR "$ASDF_DIR"
-    set -gx PATH "$ASDF_DATA_DIR/shims" $PATH
-else
-    # Fallback to old method for asdf version <= 0.15.x
-    set ASDF_SH "$ASDF_DIR/asdf.fish"
-    if test -f "$ASDF_SH"
+    set asdf_version (asdf --version)
+    set semver (string match -r '(\d+)\.(\d+)\.(\d+)' $asdf_version)
+    if test $semver[2] -eq 0 -a $semver[3] -lt 16
```

Comment:
Note to reviewer: fish is 1-indexed (not 0!) and the first thing returned is the full match, so the indexes correspond to the following:

1. 0.15.0
2. 0
3. 15
4. 0